### DISCUSSION TITLE
Beginner - Netbeans IDE 18 - Questions!

### DISCUSSION THREAD

**Discussion by valmirserpa** [apache/netbeans#6137] [Q&A] [answered]
Hello, i'm taking my first steps with Java, using Netbeans on Linux (Kali), and I have some questions.

I was using IDE version 12 and migrated to IDE 18, and I noticed some differences.

1 - Why does the class name come with "App" instead of "Main" when I create a project using Gradle in IDE 18?
2 - Can I change it to "Main" instead of "App"?
3 - Why did it create "grandleproject6" and "gradleproject6.app," and what is the difference between the two?
4 - Can I use Java with Ant  or Maven to do the same things I do with Gradle?

Thank you!

![image](https://github.com/apache/netbeans/assets/110781631/4b879b82-d321-4fec-a5d1-236f2cc392cd)
![image](https://github.com/apache/netbeans/assets/110781631/ed08b196-fac2-446d-89f5-e624ae6b32d6)
![image](https://github.com/apache/netbeans/assets/110781631/5bfeee4a-3225-48f2-89ff-f3d028c067ca)

**Comment by emaphis** [comment, marked as answer]
1,2  You can update class names and java files using Refactor/Rename in the menubar and popup menu.  Refactoring updates the class name, file name to match and updates all of the references to the class in your source code.
Check here for more hints <https://netbeans.apache.org/kb/docs/java/index.html>
3 I dont know much about Gradle but those are simply default project names applied  to files created when you create a new project.  You can create custom names when you create a new project.
4 Yes, Ant, Maven and Gradel all handle the same basic tasks but in different ways.  Consult the manual for each build tool on the internets.

HTH,
Ed

**Reply by valmirserpa** [reply]
Thanks!
